server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/CloudFoundryFileCreator.java

```diff
@@ -177,22 +174,12 @@ private boolean addMatchedServices(List<ServiceOffering> services,
         return isSet;
     }
 
-    //only for PHP 
-    //TODO: check if PHP and mysql then add "mysql" & "mysqli"
+    /**
+     detect if additional buildpacks are needed and add them
+     */
     private void createBuildpackAdditionsFile() throws IOException, JSONException {
-        JSONObject buildPackAdditionsJson = new JSONObject();
-        JSONArray buildPacks = new JSONArray();
-        for (String buildPack : app.getBuildpackAdditions()) {
-            buildPacks.put(buildPack);
-        }
-        buildPackAdditionsJson.put(BUILDPACK_OBJECT_PHP, buildPacks);
-        String path;
-        if (app.getPathToApplication() != null) {
-            path = String.format("%s/%s", app.getPathToApplication(), BUILDPACK_FILEPATH_PHP);
-        } else {
-            path = BUILDPACK_FILEPATH_PHP;
-        }
-        fileAccess.access(path).append(buildPackAdditionsJson.toString(4)).close();
+        CloudFoundryBuildpackDetection buildpackDetection = new CloudFoundryBuildpackDetection(app, fileAccess);
+        buildpackDetection.detectBuildpackAdditions();
```

Comment:
How do you know if the Buildpack addition was successfull? There is no return value neither a exception is thrown if something gs wrong.

Reply:
I check the addition in the BuildpackDetector itself (in line 42 - 46). 
Is it sufficient or do you prefer another way to do it?

Reply:
The problem is you just log if something goes wrong and the transformation does not abort.